src/algebraic_topology/dold_kan/faces.lean

```diff
@@ -0,0 +1,218 @@
+  apply simplif,
+  { /- b=f -/
+    have eq : (-1 : ℤ)^(a+1) * (-1 : ℤ)^a = -1,
+    { rw ← pow_add,
+      apply odd.neg_one_pow,
+      use a,
+      linarith, },
+    simp only [eq, neg_smul, one_zsmul], },
+  { /- d+e = 0 -/
+    let b : fin (n+2) := ⟨a+1, nat.succ_lt_succ $ nat.lt_succ_iff.mpr $
+      nat.le.intro hnaq.symm⟩,
+    have eq₁ : X.σ b ≫ X.δ (fin.cast_succ b) = 𝟙 _ := δ_comp_σ_self _,
+    have eq₂ : X.σ b ≫ X.δ b.succ = 𝟙 _ := δ_comp_σ_succ _,
+    simp only [b, fin.cast_succ_mk, fin.succ_mk] at eq₁ eq₂,
+    simp only [eq₁, eq₂, fin.last, assoc, fin.cast_succ_mk, fin.cast_le_mk, fin.coe_mk,
+      comp_id, add_eq_zero_iff_eq_neg, ← neg_zsmul],
+    congr,
+    ring_exp,
+    rw mul_one, },
+  { /- c+a = 0 -/
+    rw ← finset.sum_add_distrib,
+    apply finset.sum_eq_zero,
+    rintros ⟨i, hi⟩ h₀,
+    have hia : (⟨i, by linarith⟩ : fin(n+2)) ≤ fin.cast_succ (⟨a, by linarith⟩ : fin (n+1)) :=
+      by simpa only [fin.le_iff_coe_le_coe, fin.coe_mk, fin.cast_succ_mk, ← lt_succ_iff] using hi,
+    simp only [fin.coe_mk, fin.cast_le_mk, fin.cast_succ_mk, fin.succ_mk, assoc, fin.cast_mk,
+      ← δ_comp_σ_of_le X hia, add_eq_zero_iff_eq_neg, ← neg_zsmul],
+    congr,
+    ring_exp, },
+end
+
+lemma Hσφ_eq_zero {Y : C} {n q : ℕ} (hqn : n<q) {φ : Y ⟶ X _[n+1]}
+  (v : higher_faces_vanish q φ) : φ ≫ (Hσ q).f (n+1) = 0 :=
+begin
+  simp only [Hσ, homotopy.null_homotopic_map'_f (c_mk (n+2) (n+1) rfl) (c_mk (n+1) n rfl)],
+  erw [hσ'_eq_zero hqn (c_mk (n+1) n rfl), comp_zero, zero_add],
+  by_cases hqn' : n+1<q,
+  { rw [hσ'_eq_zero hqn' (c_mk (n+2) (n+1) rfl), zero_comp, comp_zero], },
+  { simp only [hσ'_eq (show n+1=0+q, by linarith) (c_mk (n+2) (n+1) rfl),
+      pow_zero, fin.mk_zero, one_zsmul, eq_to_hom_refl, comp_id],
+    erw chain_complex.of_d,
+    simp only [alternating_face_map_complex.obj_d, comp_sum],
+    rw [← fin.sum_congr' _ (show 2+(n+1)=n+1+2, by linarith), fin.sum_trunc],
+    { simp only [fin.sum_univ_cast_succ, fin.sum_univ_zero, zero_add, fin.last,
+        fin.cast_le_mk, fin.cast_mk, fin.cast_succ_mk],
+      simp only [fin.mk_zero, fin.coe_zero, pow_zero, one_zsmul, fin.mk_one,
+        fin.coe_one, pow_one, neg_smul, comp_neg],
+      erw [δ_comp_σ_self, δ_comp_σ_succ, add_right_neg], },
+    { intro j,
+      simp only [comp_zsmul],
+      convert zsmul_zero _,
+      have h : fin.cast (by rw add_comm 2) (fin.nat_add 2 j) = j.succ.succ,
+      { ext, simp only [add_comm 2, fin.coe_cast, fin.coe_nat_add, fin.coe_succ], },
+      rw [h, ← fin.cast_succ_zero, δ_comp_σ_of_gt X], swap,
+      { exact fin.succ_pos j, },
+      simp only [← assoc, v j (by linarith), zero_comp], }, },
+end
+
+lemma higher_faces_vanish_induction {Y : C} {n q : ℕ} {φ : Y ⟶ X _[n+1]}
+  (v : higher_faces_vanish q φ) : higher_faces_vanish (q+1) (φ ≫ (𝟙 _ + Hσ q).f (n+1)) :=
+begin
+  intros j hj₁,
+  simp only [add_comp, comp_add, homological_complex.add_f_apply, homological_complex.id_f],
+  erw comp_id,
```

Comment:
I don't think this should be necessary. Could you change the `@[simps]` on `alternating_face_map_complex ` to `@[simps?]` and check if these are the ideal simp lemmas? I think it might be better if `alternating_face_map_complex.obj` was being unfolded to `chain_complex.of (λ n, X _[n]) (obj_d X) (d_squared X)`.

I appreciate that there are a lot of `erw`s needed in the category theory library, but I think it's worth investigating each new appearance --- often they are sign that definitionally true simplifications of objects aren't quite set up correctly.

Ping me if this doesn't make sense. :-)

Reply:
Thanks for the review!
I have added simp lemmas for `alternating_face_map_complex` which should be more practical. However, I did not manage to get `𝟙 (((alternating_face_map_complex C).obj X).X (n + 1)))` simped into `𝟙 (X _[n+1])`. If I do `dsimp only [alternating_face_map_complex, alternating_face_map_complex.obj, chain_complex.of]`, then `rw comp_id` shall work, otherwise, only `erw comp_id` works.

Reply:
Except for the fact that notation is mysteriously not being used in the output, itsn't it dsimping into exactly that at the moment?

Reply:
I have removed most of the `erw` that were left.